config

```diff
@@ -1,926 +1,10 @@
-	OBJECT_MODE=${OBJECT_MODE:-32}
-	if [ "$CC" = "gcc" ]; then
-	    OUT="aix-gcc"
-          if [ $OBJECT_MODE -eq 64 ]; then
-            echo 'Your $OBJECT_MODE was found to be set to 64'
-            OUT="aix64-gcc"
-          fi
-	elif [ $OBJECT_MODE -eq 64 ]; then
-	    echo 'Your $OBJECT_MODE was found to be set to 64'
-	    OUT="aix64-cc"
-	else
-	    OUT="aix-cc"
-	    if [ $KERNEL_BITS -eq 64 ]; then
-		echo "WARNING! If you wish to build 64-bit kit, then you have to"
-		echo "         invoke '$THERE/Configure aix64-cc' *manually*."
-		if [ "$DRYRUN" = "false" -a -t 1 ]; then
-		    echo "         You have ~5 seconds to press Ctrl-C to abort."
-		    (trap "stty `stty -g`; exit 0" 2 0; stty -icanon min 0 time 50; read waste) <&1
-		fi
-	    fi
-	fi
-	if (lsattr -E -O -l `lsdev -c processor|awk '{print$1;exit}'` | grep -i powerpc) >/dev/null 2>&1; then
-	    :	# this applies even to Power3 and later, as they return PowerPC_POWER[345]
-	else
-	    options="$options no-asm"
-	fi
-	;;
-  # these are all covered by the catchall below
-  i[3456]86-*-cygwin) OUT="Cygwin-x86" ;;
-  *-*-cygwin) OUT="Cygwin-${MACHINE}" ;;
-  x86-*-android|i?86-*-android) OUT="android-x86" ;;
-  armv[7-9]*-*-android)
-      OUT="android-armeabi"
-      __CNF_CFLAGS="$__CNF_CFLAGS -march=armv7-a"
-      __CNF_CXXFLAGS="$__CNF_CXXFLAGS -march=armv7-a";;
-  arm*-*-android) OUT="android-armeabi" ;;
-  *) OUT=`echo $GUESSOS | awk -F- '{print $3}'`;;
-esac
-
-# NB: This atalla support has been superseded by the ENGINE support
-# That contains its own header and definitions anyway. Support can
-# be enabled or disabled on any supported platform without external
-# headers, eg. by adding the "hw-atalla" switch to ./config or
-# perl Configure
-#
-# See whether we can compile Atalla support
-#if [ -f /usr/include/atasi.h ]
-#then
-#  __CNF_CPPFLAGS="$__CNF_CPPFLAGS -DATALLA"
-#fi
-
-if [ -n "$CONFIG_OPTIONS" ]; then
-  options="$options $CONFIG_OPTIONS"
-fi
-
-# gcc < 2.8 does not support -march=ultrasparc
-if [ "$OUT" = solaris-sparcv9-gcc -a $GCCVER -lt 28 ]
-then
-  echo "WARNING! Falling down to 'solaris-sparcv8-gcc'."
-  echo "         Upgrade to gcc-2.8 or later."
-  sleep 5
-  OUT=solaris-sparcv8-gcc
-fi
-if [ "$OUT" = "linux-sparcv9" -a $GCCVER -lt 28 ]
-then
-  echo "WARNING! Falling down to 'linux-sparcv8'."
-  echo "         Upgrade to gcc-2.8 or later."
-  sleep 5
-  OUT=linux-sparcv8
-fi
-
-case "$GUESSOS" in
-  i386-*) options="$options 386" ;;
-esac
-
-if [ -z "$OUT" ]; then
-  OUT="$CC"
-fi
-
-if [ ".$PERL" = . ] ; then
-	for i in . `echo $PATH | sed 's/:/ /g'`; do
-		if [ -f "$i/perl5$EXE" ] ; then
-			PERL="$i/perl5$EXE"
-			break;
-		fi;
-	done
-fi
-
-if [ ".$PERL" = . ] ; then
-	for i in . `echo $PATH | sed 's/:/ /g'`; do
-		if [ -f "$i/perl$EXE" ] ; then
-			if "$i/perl$EXE" -e 'exit($]<5.0)'; then
-				PERL="$i/perl$EXE"
-				break;
-			fi;
-		fi;
-	done
-fi
-
-if [ ".$PERL" = . ] ; then
-	echo "You need Perl 5."
-	exit 1
-fi
```

Comment:
Would it be still useful to keep some perl checks here as this is a shell script? For example check whether there is any perl at all and whether it has the minimum version supported? Or are we OK with just dropping this? There is Perl 5 mentioned in INSTALL.md prerequisities but I suppose there is some minimum 5.x version needed.

Reply:
If there is no perl at all, the `exec` line will fail and tell us so.
If the version isn't good enough, there's a line `use 5.10.0;` early in Configure that will tell us so.

Note that the checks above will not at all check the user given `$PERL`, not even if it's the proper version.

So essentially, I think we have all bases covered, we're just using the proper mechanisms for it instead of replicating it ourselves.

Reply:
For the sake of experimenting, I modified `use 5.10.0` to `use 5.40.0`:

``` console
$ ./config
Perl v5.40.0 required--this is only v5.30.3, stopped at /home/levitte/gitwrk/openssl.net/official/master/Configure line 12.
BEGIN failed--compilation aborted at /home/levitte/gitwrk/openssl.net/official/master/Configure line 12.
```

Reply:
Good!